nexus-client/src/lib.rs

```diff
@@ -99,63 +78,22 @@ impl From<omicron_common::api::internal::nexus::InstanceRuntimeState>
     }
 }
 
-impl From<&omicron_common::api::internal::nexus::InstanceRuntimeState>
-    for types::InstanceRuntimeState
-{
-    fn from(
-        s: &omicron_common::api::internal::nexus::InstanceRuntimeState,
-    ) -> Self {
-        Self {
-            run_state: s.run_state.into(),
-            sled_uuid: s.sled_uuid,
-            propolis_uuid: s.propolis_uuid,
-            dst_propolis_uuid: s.dst_propolis_uuid,
-            propolis_addr: s.propolis_addr.map(|addr| addr.to_string()),
-            migration_uuid: s.migration_uuid,
-            ncpus: s.ncpus.into(),
-            memory: s.memory.into(),
-            hostname: s.hostname.clone(),
-            gen: s.gen.into(),
-            time_updated: s.time_updated,
-        }
-    }
-}
-
 impl From<omicron_common::api::external::InstanceState>
     for types::InstanceState
 {
     fn from(s: omicron_common::api::external::InstanceState) -> Self {
+        use omicron_common::api::external::InstanceState;
         match s {
-            omicron_common::api::external::InstanceState::Creating => {
-                Self::Creating
-            }
-            omicron_common::api::external::InstanceState::Starting => {
-                Self::Starting
-            }
-            omicron_common::api::external::InstanceState::Running => {
-                Self::Running
-            }
-            omicron_common::api::external::InstanceState::Stopping => {
-                Self::Stopping
-            }
-            omicron_common::api::external::InstanceState::Stopped => {
-                Self::Stopped
-            }
-            omicron_common::api::external::InstanceState::Rebooting => {
-                Self::Rebooting
-            }
-            omicron_common::api::external::InstanceState::Migrating => {
-                Self::Migrating
-            }
-            omicron_common::api::external::InstanceState::Repairing => {
-                Self::Repairing
-            }
-            omicron_common::api::external::InstanceState::Failed => {
-                Self::Failed
-            }
-            omicron_common::api::external::InstanceState::Destroyed => {
-                Self::Destroyed
-            }
+            InstanceState::Creating => Self::Creating,
+            InstanceState::Starting => Self::Starting,
+            InstanceState::Running => Self::Running,
+            InstanceState::Stopping => Self::Stopping,
+            InstanceState::Stopped => Self::Stopped,
+            InstanceState::Rebooting => Self::Rebooting,
+            InstanceState::Migrating => Self::Migrating,
+            InstanceState::Repairing => Self::Repairing,
+            InstanceState::Failed => Self::Failed,
+            InstanceState::Destroyed => Self::Destroyed,
```

Comment:
do we even need these still? I bet there's a way to get rid of them. I'll take a look

Reply:
So this PR is admittedly mostly focused on the "pieces of sled agent in `common`" side of things, and hasn't attempted to remove the "pieces of nexus in `common`".

The top-level structure being converted here is `InstanceRuntimeState` (which contains `InstanceState`).

I expected that the dependencies on `omicron_common::api::internal::nexus` to be migrated to either using `nexus::params` or `nexus_client::types` as a separate patch, which would remove the need for the conversion here.

Reply:
Sounds great!